lib/core/CDataFrame.cc

```diff
@@ -495,13 +495,29 @@ std::size_t CDataFrame::estimateMemoryUsage(bool inMainMemory,
                                             std::size_t numberRows,
                                             std::size_t numberColumns,
                                             CAlignment::EType alignment) {
-    return sizeof(CDataFrame) + core::memory::dynamicSize(TStrVec(numberColumns)) +
-           core::memory::dynamicSize(TStrVecVec(numberColumns)) +
-           core::memory::dynamicSize(TStrSizeUMapVec(numberColumns)) +
-           core::memory::dynamicSize(TBoolVec(numberColumns)) +
-           (inMainMemory ? numberRows * CAlignment::roundupSizeof<CFloatStorage>(alignment, numberColumns)
-                         : 0);
-    ;
+    std::size_t estimatedMemoryUsage =
+        sizeof(CDataFrame) + core::memory::dynamicSize(TStrVec(numberColumns)) +
+        core::memory::dynamicSize(TStrVecVec(numberColumns)) +
+        core::memory::dynamicSize(TStrSizeUMapVec(numberColumns)) +
+        core::memory::dynamicSize(TBoolVec(numberColumns)) +
+        (inMainMemory ? numberRows * CAlignment::roundupSizeof<CFloatStorage>(alignment, numberColumns)
+                      : 0);
+
+    // Beginning with Boost 1.80, unordered containers require more memory per
+    // bucket than those of previous versions. Even though this has largely been
+    // accounted for in our memory estimations there still is a small discrepancy
+    // between actual and estimated memory. We account for that maximum percentage
+    // difference here in boostUnorderedContainerMemorySlackPercent.
+    static constexpr double boostUnorderedContainerMemorySlackPercent{2.5};
```

Comment:
I think there was also underestimation already in the `TStrVecVec` value, since the existing estimate assumes all the nested vectors are empty. It's probably worth making this fudge factor less specific to the Boost changes, and call the variable something more generic.